classes/statistics/extStatistics.sc

```diff
@@ -480,6 +480,91 @@ oscorr(x,y);
 		};
 		^p
 	}
+
+	// Simple Linear Regression (Least Squares) - Linear Fit
+	linearFit {
+		var tmpThis;
+		var size, x, y;
+		var xMean, yMean, sSxy, sSxx, b0, b1;
+		var errStr = "Support is only provided for 2D line fitting. Supply y or x, y pairs only.";
+		var err = Error.new(errStr);
+
+		size = this.size;
+		this.rank.switch(
+			1, { x = Array.series(size); y = this },
+			2, {
+				(this.first.size == 2).if({
+					tmpThis = this.flop; x = tmpThis.at(0); y = tmpThis.at(1)
+				}, {
+					err.throw
+				})
+			},
+			{ err.throw },
+		);
+
+		// cast as DoubleArray for precision
+		x = x.as(DoubleArray);
```

Comment:
My assumption is that using `DoubleArray` is an easy way to make sure everything is 64 bit floats.

For `-linearFit`, if either x or y is a collection of `Integer`, then it is easy to overflow and return the wrong answer with large data sets. (I kept the same thing for `theilSenFit`, as a precaution.)

Would you suggest doing something different?

Reply:
Well, it makes it impossible to work with arrays of floats instead of floats, or any new class that is compatible with the math operations. Without the conversion, that should work out of the box, when I read your code. It could even work with an array of patterns or an array of UGens.

I'd just create the array as an array of floats: `Array.series(100, 0.0, 1)`

Reply:
Ok... actually, it looks like negligence on my part. I should do the casting as `x.asFloat` and `y.asFloat`.

> I'd just create the series of floats directly: `Array.series(100, 0.0, 1)`

As you say, this will work for `x` if automagically generated, but we also need to catch `y`, which will always be supplied.

Reply:
Ok... casting to float here: 09122d09609e6d8d68635d2df106907cfd748076

You'll notice I have to catch `Signal` as `-asFloat` throws an error....